### DISCUSSION TITLE
Is it possible to add any components before or after the collection's List component?

### DISCUSSION THREAD

**Discussion by wherewego-nina** [payloadcms/payload#2746] [Feature Requests & Ideas]
I would like to add functionality to export a csv of my data. From my searching, it looks like Payload hasn't exposed the List components yet so that I could import them into a custom component, so I'm wondering if there is a mechanism for inserting a component before or after the List component

**Comment by DanRibbens** [comment]
Hey @wherewego-nina,

This is not currently possible and is something we want to address right away. The simplest change we can make to support you would be to introduce `beforeList` and `afterList` as new arrays which will hold components to render for the list view. We do this already on the dashboard and a few other places and should work the same. Since these may need to be collection specific, these properties should be added to collection `admin.components` rather than the top `admin` for more granularity. We should also export the default list component that Payload uses internally to make it easier to work with without having to replace it (this is currently has to be done by importing from `/dist` which works but is more suseptible to breaking changes. That way you could wrap it in another component, display it dynamically or not at all if you wanted.

These two changes can be done in short order.

An alternate way of doing this would be to add `actions` as a new collection feature. With this proposal, We'd introduce a place in the configuration with an array of objects for managing the UI and endpoints used to complete actions. I believe your use-case is a perfect example of where a general `actions` option would be useful.

I've started a new discusson to outline the `actions` proposal since I can't find where we have talked about it before. https://github.com/payloadcms/payload/discussions/2755

**Reply by wherewego-nina** [reply]
Yes, I think maybe all three would be great additions!

For `beforeList` and `afterList`, I agree that granularity would be great. It would be awesome to have access to the same props as the Table in each of those components as well. 

For now, I'll import from `/dist` because it's a pretty urgent issue for me, but I'd love to see that exposed internally. 

I love the idea of actions! Seems like a natural extension of table functionality. Thank you for the quick response!
